src/m365/aad/commands/user/user-remove.spec.ts

```diff
@@ -0,0 +1,208 @@
+import commands from '../../commands';
+const command: Command = require('./user-remove');
+
+describe(commands.USER_REMOVE, () => {
+  let commandInfo: CommandInfo;
+  //#region Mocked Responses
+  const validId = '3a081d91-5ea8-40a7-8ac9-abbaa3fcb893';
+  const validUsername = 'john.doe@contoso.com';
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let promptOptions: any;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.USER_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if id is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        id: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { userName: validUsername } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified user when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        id: validId
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified user when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'delete');
```

Comment:
In fact, it's correct to use a spy here. But imagine that our prompt doesn't work, in that case it will try to remove the user even if we didn't confirm. Because we have no `delete` stub, it will timeout after 10 seconds. This will result in an error `The test timed out after 10 000ms`. In that case, the message of the test will not be clear about what is actually wrong.

If we create a stub for `request.delete(...)`, the test will not timeout and we'll get an error like `expected: true, actual: false`. In that case it's immediately clear for us that something is wrong with `postSpy.notCalled`.
While writing this, I also realized we should replace `post` by `delete` in the variable name 😄